types/later/index.d.ts

```diff
@@ -146,7 +146,8 @@ declare namespace later {
          * @param dateFrom: The earliest a valid instance can occur
          * @param dateTo: The latest a valid instance can occur
          */
-        next(numberOfInst: number, dateFrom?: Date, dateTo?: Date): Date[] | Date;
+        next(numberOfInst: 1, dateFrom?: Date, dateTo?: Date): Date;
+        next(numberOfInst: number, dateFrom?: Date, dateTo?: Date): Date[];
```

Comment:
`1` also matches `number`. [Playground example](https://www.typescriptlang.org/play/index.html?ssl=1&ssc=1&pln=10&pc=32#code/DYUwLgBAbghsCuIIF4IEYDcAoLBLAdmCAE4BmMAxkgMoUAWIAJvKBAN5YRcT4gAeYABT54AWwBGJAPKkAkvgDOYAFzoANBEYwiAMWIB7UQH5VAEW0gNWogBV9JiOaIBKMxezce-ISInS5iio8YpLEVhZ6hg5OlpoWdtEWro4WANoAutgAvjiMIBTAMMRIFPqBEDCqtAzMoNhYMAB0vAKCsAggzo0AVvoEgs4YEAD0wxAA7gb4AOYAhEA).

```ts
let value = 1;

interface Schedule {
    next(numberOfInst: 1, dateFrom?: Date, dateTo?: Date): Date;
    next(numberOfInst: number, dateFrom?: Date, dateTo?: Date): Date[];
}

declare const a: Schedule;

a.next(value).join(); // wrong!
```

I suggest just adding overload for 1.

```suggestion
        next(numberOfInst: 1, dateFrom?: Date, dateTo?: Date): Date;
        next(numberOfInst: number, dateFrom?: Date, dateTo?: Date): Date[] | Date;	
```

Reply:
indeed 👌